addons/alerting.libsonnet

```diff
@@ -10,7 +10,159 @@ function(config) {
     "If 'alerting' is set, 'slackWebhookURLCritical' or 'pagerdutyRoutingKey' should be declared"
   ),
 
-  local criticalReceiver =
+  local IDECriticalReceiver =
+    if
```

Comment:
This would be a little bit more readable if we did something like

```jsonnet
local globalPagerDutyRoutingKey = if std.objectHas(config.alerting, 'pagerdutyRoutingKey')
  then config.alerting.pagerdutyRoutingKey
  else ""

local IDEPagerDutyRoutingKey = if std.objectHas(config.alerting, 'IDE') && std.objectHas(config.alerting.IDE, 'pagerdutyRoutingKey') 
  then config.alerting.IDE.pagerdutyRoutingKey
  else globalPagerDutyRoutingKey

local IDECriticalReceiver =
  if std.length(IDEPagerDutyRoutingKey) > 0
  then 
    |||
        pagerduty_configs:
          - send_resolved: true
            routing_key: '%(pagerdutyRoutingKey)s'
    ||| % {
        pagerdutyRoutingKey: IDEPagerDutyRoutingKey
    }
  else
    // Slack goes here
```

Reply:
Ah right, reading the code it looks like that was the case before too. Can we just simplify it and say

1. If there is an IDE pagerdutyRoutingKey then use PD
2. Otherwise use Slack (using the team slack, falling back to global slack)

Something like

```jsonnet
local IDEPagerDutyRoutingKey = if std.objectHas(config.alerting, 'IDE') && std.objectHas(config.alerting.IDE, 'pagerdutyRoutingKey') 
  then config.alerting.IDE.pagerdutyRoutingKey
  else ""

local IDECriticalReceiver =
  if std.length(IDEPagerDutyRoutingKey) > 0
  then 
    |||
        pagerduty_configs:
          - send_resolved: true
            routing_key: '%(pagerdutyRoutingKey)s'
    ||| % {
        pagerdutyRoutingKey: IDEPagerDutyRoutingKey
    }
  else
    // Slack goes here
```
```

Reply:
Ah wait, I finally see the subtle distinction that if there wasn't an config.alerting.IDE object at all it would use the global PD key ☺️

I'm still leaning towards going with the suggestion before though (that if you configure an alert for a team but you don't have any alerting config for the team, then it just goes to slack) just to keep the logic simple

Reply:
@ArthurSens Thanks for the loom, looking at the code you have open it's missing a minor change I made in the comment above (that the "else" branch returns "" instead of the global routing key) ☺️ I'm about to head out but if I'm still misunderstanding something I will try to be online tonight and perhaps we can have a quick sync call instead ☺️